modules/base/rendering/renderableswitch.h

```diff
@@ -0,0 +1,66 @@
+ * Copyright (c) 2014-2025                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_MODULE_BASE___RENDERABLESWITCH___H__
+#define __OPENSPACE_MODULE_BASE___RENDERABLESWITCH___H__
+
+#include <modules/base/rendering/RenderablePlaneImageLocal.h>
+#include <modules/base/rendering/RenderablePlaneImageOnline.h>
+#include <modules/base/rendering/renderableplane.h>
+#include <openspace/rendering/renderable.h>
+
+#include <openspace/util/updatestructures.h>
+
+
+namespace openspace {
+
+    struct RenderData;
+    struct UpdateData;
+
+    namespace documentation { struct Documentation; }
+
+    class RenderableSwitch : public Renderable {
+    public:
+        RenderableSwitch(const ghoul::Dictionary& dictionary);
+
+        void initializeGL() override;
+        void deinitializeGL() override;
+
+        bool isReady() const override;
+
+        void update(const UpdateData& data) override;
+        void render(const RenderData& data, RendererTasks& tasks) override;
+
+        static documentation::Documentation Documentation();
+
+    protected:
+        properties::BoolProperty _autoScale;
+        ghoul::mm_unique_ptr<Renderable> _renderable1;
+        ghoul::mm_unique_ptr<Renderable> _renderable2;
+        properties::FloatProperty _distanceThreshold;
```

Comment:
```suggestion
        properties::DoubleProperty _distanceThreshold;
```

Since we might want to use quite large distances, storing it in a `double` might be beneficial.